uniffi_bindgen/src/pipeline/general/nodes.rs

```diff
@@ -112,15 +112,12 @@ pub enum CallableKind {
     /// Interface/Trait interface method
     Method { self_type: TypeNode },
     /// Interface constructor
-    Constructor {
-        interface_name: String,
-        primary: bool,
-    },
+    Constructor { self_type: TypeNode, primary: bool },
```

Comment:
I thought I needed these `self_type` changes - turns out I don't but I think it's an improvement wrt consistency?